src/components/Main/state/caseCountsData.ts

```diff
@@ -39,3 +40,9 @@ export function getCaseCountsData(key: string) {
   const [caseCount] = Convert.toCaseCounts(JSON.stringify([caseCountFound]))
   return caseCount.empiricalData
 }
+
+export function getSortedNonEmptyCaseCounts(key: string): EmpiricalData {
+  return getCaseCountsData(key)
+    .filter((d) => d.cases || d.deaths || d.icu || d.hospitalized)
+    .sort((a, b) => (a.time > b.time ? 1 : -1))
+}
```

Comment:
The core part of the refactoring: groups the functionality together in one place, under a meaningful function name.